// hv-color/hv/transavia.js
const hex = {
  Green: "#00d66c",
  GreenDark: "#1e9162",
  GreenLight: "#4de298",
  Ruby: "#ff3268",
  RubyDark: "#ca2b55",
  RubyLight: "#ff658e",
  Blue: "#2800a0",
  BlueDark: "#120145",
  BlueLight: "#5e40b8",
  Champagne: "#f3e5d8",
  GrayDarkest: "#202020",
  GrayDarker: "#606060",
  GrayDark: "#909090",
  GrayMid: "#cccccc",
  GrayLight: "#e6e6e6",
  GrayLigher: "#f7f7f7",
  GrayLightest: "#f0f0f0",
  Black: "#000000",
  White: "#ffffff"
};

const hsl = {
  Green: {
    h: 150.2803738317757,
    s: 100,
    l: 41.96078431372549
  },
  GreenDark: {
    h: 155.47826086956522,
    s: 65.71428571428571,
    l: 34.31372549019608
  },
  GreenLight: {
    h: 150.2013422818792,
    s: 71.9806763285024,
    l: 59.411764705882355
  },
  Ruby: {
    h: 344.1951219512195,
    s: 100,
    l: 59.80392156862745
  },
  RubyDark: {
    h: 344.1509433962264,
    s: 64.89795918367346,
    l: 48.0392156862745
  },
  RubyLight: {
    h: 344.02597402597405,
    s: 100,
    l: 69.80392156862744
  },
  Blue: {
    h: 255,
    s: 100,
    l: 31.372549019607842
  },
  BlueDark: {
    h: 255,
    s: 97.14285714285715,
    l: 13.72549019607843
  },
  BlueLight: {
    h: 255,
    s: 48.38709677419355,
    l: 48.627450980392155
  },
  Champagne: {
    h: 28.888888888888925,
    s: 52.94117647058819,
    l: 89.99999999999999
  },
  GrayDarkest: {
    h: 0,
    s: 0,
    l: 12.549019607843137
  },
  GrayDarker: {
    h: 0,
    s: 0,
    l: 37.64705882352941
  },
  GrayDark: {
    h: 0,
    s: 0,
    l: 56.470588235294116
  },
  GrayMid: {
    h: 0,
    s: 0,
    l: 80
  },
  GrayLight: {
    h: 0,
    s: 0,
    l: 90.19607843137256
  },
  GrayLigher: {
    h: 0,
    s: 0,
    l: 96.86274509803921
  },
  GrayLightest: {
    h: 0,
    s: 0,
    l: 94.11764705882352
  },
  Black: {
    h: 0,
    s: 0,
    l: 0
  },
  White: {
    h: 0,
    s: 0,
    l: 100
  }
};

const rgb = {
  Green: {
    r: 0,
    g: 214,
    b: 108
  },
  GreenDark: {
    r: 30,
    g: 145,
    b: 98
  },
  GreenLight: {
    r: 77,
    g: 226,
    b: 152
  },
  Ruby: {
    r: 255,
    g: 50,
    b: 104
  },
  RubyDark: {
    r: 202,
    g: 43,
    b: 85
  },
  RubyLight: {
    r: 255,
    g: 101,
    b: 142
  },
  Blue: {
    r: 40,
    g: 0,
    b: 160
  },
  BlueDark: {
    r: 18,
    g: 1,
    b: 69
  },
  BlueLight: {
    r: 94,
    g: 64,
    b: 184
  },
  Champagne: {
    r: 243,
    g: 229,
    b: 216
  },
  GrayDarkest: {
    r: 32,
    g: 32,
    b: 32
  },
  GrayDarker: {
    r: 96,
    g: 96,
    b: 96
  },
  GrayDark: {
    r: 144,
    g: 144,
    b: 144
  },
  GrayMid: {
    r: 204,
    g: 204,
    b: 204
  },
  GrayLight: {
    r: 230,
    g: 230,
    b: 230
  },
  GrayLigher: {
    r: 247,
    g: 247,
    b: 247
  },
  GrayLightest: {
    r: 240,
    g: 240,
    b: 240
  },
  Black: {
    r: 0,
    g: 0,
    b: 0
  },
  White: {
    r: 255,
    g: 255,
    b: 255
  }
};

export default { hex, rgb, hsl };
